core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -231,6 +243,12 @@ public void setMaster(@Nullable SchemaRegistryIdentity newMaster)
         masterRestService = null;
       } else {
         masterRestService = new RestService(masterIdentity.getUrl());
+        if ("https".equals(masterIdentity.getScheme()) && !masterRestSslInitialized) {
+          SslFactory sslFactory = new SslFactory();
```

Comment:
Since this isn't dependent on the specific master, why not just do this up front? If it's not configured, it won't be initialized; if it is configured, they probably want it so any potential overhead doesn't seem like a big deal.